packages/core/scripts/admin-proposals/identifier.js

```diff
@@ -0,0 +1,278 @@
+// Run:
+// - For testing, start mainnet fork in one window with `yarn hardhat node --fork <ARCHIVAL_NODE_URL> --no-deploy`
+// - (optional, or required if --polygon is not undefined) set CROSS_CHAIN_NODE_URL to a Polygon mainnet node. This will
+// be used to query contract data from Polygon when relaying proposals through the GovernorRootTunnel.
+// - Propose: HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/identifier.js --identifier 0xabc,0x123 --polygon 0xabc,0x123
+// - Vote Simulate: HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/simulateVote.js
+// - Verify: HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/identifier.js --verify --polygon 0xabc,0x123 --identifier 0xabc,0x123
+// - For production, set the CUSTOM_NODE_URL environment and run the script with a different `HARDHAT_NETWORK` value,
+//   for example: `HARDHAT_NETWORK=mainnet node ./packages/core/scripts/admin-proposals/identifier.js ...`
+
+// Customizations:
+// - --polygon param can be omitted, in which case transactions will only take place on Ethereum.
+// - --identifier flag can also be omitted, in which case transactions will only be relayed to Polygon
+// - --identifier and --polygon params must be comma delimited strings producing equal length array of identifiers to approve
+// - Specific identifier or polygon-identifiers can be skipped to conform with the above constraint like so: --identifier ,, --polygon 0xab,,
+// - If --verify flag is set, script is assumed to be running after a Vote Simulation and updated contract state is
+// verified.
+
+// Examples:
+// - Whitelist identifiers on Ethereum only:
+//    - `HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/identifier.js --identifier "POOL/USD","USD/POOL"`
+// - Whitelist identifiers on Polygon only:
+//    - `HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/identifier.js --polygon "POOL/USD","USD/POOL"`
+// - Whitelist identifiers on both (some on Ethereum, some on Polygon):
+//    - `HARDHAT_NETWORK=localhost node ./packages/core/scripts/admin-proposals/identifier.js --identifier "POOL/USD","USD/POOL" --polygon "POOL/USD",`
+
+const hre = require("hardhat");
+require("dotenv").config();
+const { GasEstimator } = require("@uma/financial-templates-lib");
+const Web3 = require("web3");
+const winston = require("winston");
+const { interfaceName } = require("@uma/common");
+const { _getContractAddressByName, _impersonateAccounts } = require("./utils");
+const argv = require("minimist")(process.argv.slice(), {
+  string: [
+    // comma-delimited list of identifiers to whitelist on Ethereum. Required if --polygon is omitted.
+    "identifier",
+    // comma-delimited list of identifiers to whitelist on Polygon. Required if --identifier is omitted.
+    "polygon",
+  ],
+  boolean: [
+    // set True if verifying, False for proposing.
+    "verify",
```

Comment:
no point in changing it now but it might have been a bit clenaer to make a proposal and a verify script separately rather than having any branching logic.

Reply:
I disagree actually. Would be a ton more code since there is a lot of shared contract setup.